app/frontend/actions.js

```diff
@@ -385,29 +433,30 @@ module.exports = ({setState, getState}) => {
       sendResponse: '',
       sendAmount: Object.assign({}, state.sendAmount, {
         fieldValue: e.target.value,
+        coins: parseCoins(e.target.value),
       }),
     })
     validateSendFormAndCalculateFee()
   }
 
   const sendMaxFunds = async (state) => {
     setState({calculatingFee: true})
-
-    const maxAmount = await wallet.getMaxSendableAmount(state.sendAddress.fieldValue)
-
-    if (maxAmount > 0) {
+    let maxAmount
+    try {
+      maxAmount = await wallet.getMaxSendableAmount(state.sendAddress.fieldValue)
       setState({
         sendResponse: '',
-        sendAmount: sendAmountValidator(printAda(maxAmount)),
+        sendAmount: {
+          fieldValue: printAda(maxAmount),
+          coins: maxAmount || null,
```

Comment:
before the changes, maxAmount was being checked if it is grater than zero. Are you sure removing it is safe?

Reply:
I am quite sure, first of all, the max amount cant be less then 0, check `getMaxSendableAmount` function in `cardano-wallet.js`, the least amount it can return is 0 in which case the maxAmount is set to `null`. Secondly, the amount gets validated either way so there is no way, negative value would pass.